openedx/core/djangoapps/catalog/tests/test_utils.py

```diff
@@ -17,14 +19,108 @@
     get_programs_with_type,
     get_course_runs,
 )
-from openedx.core.djangolib.testing.utils import skip_unless_lms
+from openedx.core.djangolib.testing.utils import CacheIsolationTestCase, skip_unless_lms
 from student.tests.factories import UserFactory
 
+
 UTILS_MODULE = 'openedx.core.djangoapps.catalog.utils'
 User = get_user_model()  # pylint: disable=invalid-name
 
 
 @skip_unless_lms
+@mock.patch(UTILS_MODULE + '.logger.warning')
+class TestGetCachedPrograms(CacheIsolationTestCase):
+    ENABLED_CACHES = ['default']
+
+    def setUp(self):
+        super(TestGetCachedPrograms, self).setUp()
+
+        Switch.objects.create(name='read_cached_programs', active=True)
+
+    def test_get_many(self, mock_warning):
+        programs = ProgramFactory.create_batch(3)
+
+        # Cache details for 2 of 3 programs.
+        partial_programs = {
+            PROGRAM_CACHE_KEY_TPL.format(uuid=program['uuid']): program for program in programs[:2]
+        }
+        cache.set_many(partial_programs, None)
+
+        # When called before UUIDs are cached, the function should return an empty
+        # list and log a warning.
+        self.assertEqual(get_programs(), [])
+        mock_warning.assert_called_once_with('Program UUIDs are not cached.')
+        mock_warning.reset_mock()
+
+        # Cache UUIDs for all 3 programs.
```

Comment:
UUIDs for all 3 programs are cached, but details are only cached for 2. This covers the case where details for a program might be missing even though its UUID is included in the list of known programs.